test/framework/seedregistrationframework.go

```diff
@@ -0,0 +1,359 @@
+package framework
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/onsi/ginkgo"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var seedCreationConfig *SeedRegistrationConfig
+
+const (
+	kubeconfigString  = "kubeconfig"
+	accesKeyString    = "secretAccessKey"
+	accessKeyIdString = "accessKeyID"
+)
+
+// SeedRegistrationFramework represents the seed test framework that includes
+// test functions that can be executed on a specific seed
+type SeedRegistrationFramework struct {
+	*GardenerFramework
+	TestDescription
+	Config *SeedRegistrationConfig
+
+	SeedClient kubernetes.Interface
+	Seed       *gardencorev1beta1.Seed
+	Secret     *corev1.Secret
+}
+
+// SeedRegistrationConfig is the configuration for a seed registration framework that will be filled with user provided data
+type SeedRegistrationConfig struct {
+	GardenerConfig        *GardenerConfig
+	SeedName              string
+	AccessKeyId           string
```

Comment:
I think we should not add provider specific data in the gardener framework.
We should rather work with secrets or more generic data.

Reply:
Yep, this is just for the draft PR.